corehq/apps/domain/test_decorators.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from django.test import TestCase
+from django.contrib.auth import get_user_model
+from django.test.client import Client
+from corehq.util.test_utils import flag_enabled
+from corehq.apps.users.models import CouchUser
+from corehq.apps.domain.shortcuts import create_domain
+from corehq.apps.domain.decorators import _two_factor_required
+
+class TestDecorators(TestCase):
+
+    def setUp(self):
+        self.domain = create_domain("test_domain")
+        self.domain.two_factor_auth = False
+        self.request_2fa_enabled = self.create_request(request_url="/account/",
+                                                       username="test0@test.com",
+                                                       password="123",
+                                                       )
+        self.request_2fa_disabled = self.create_request(request_url="/account/",
+                                                       username="test_1@test.com",
+                                                       password="123",
+                                                       )
+
+    def create_request(self, request_url, username, password):
+        # Initialize request
+        request = Client().get(request_url).wsgi_request
+        # Create user
+        request.user = get_user_model().objects.create_user(username=username, email=username,
+                                                            password=password)
+        # Create couch user
+        request.couch_user = CouchUser()
+        # Login
+        assert Client().login(username=username, password=password)
+        return request
```

Comment:
The way you've written your tests, I don't think you need any of this... you can just pass in the `CouchUser` to the function you are testing.

Reply:
My new tests need request.couch_user, so I changed this to just initialize a request and creat a couch user, unneeded stuff is deleted here:

https://github.com/dimagi/commcare-hq/pull/19570/commits/474935b1ae27265ae264bcc16efb7a2fa2102efd